node/node.go

```diff
@@ -638,11 +652,11 @@ func (node *Node) handleNewHeaviestTipSet(ctx context.Context, head types.TipSet
 				continue
 			}
 
-			// When a new best TipSet is promoted we remove messages in it from the
-			// message pool (and add them back in if we have a re-org).
+			if err := outboxPolicy.OnNewHeadTipset(ctx, head, newHead); err != nil {
+				log.Error("updating outbound message queue for new tipset", err)
+			}
 			if err := node.MsgPool.UpdateMessagePool(ctx, node.ChainReadStore(), head, newHead); err != nil {
-				log.Error("error updating message pool for new tipset:", err)
-				continue
```

Comment:
Yes, for just the reasons you stated. It's an error-level problem if the message pool update fails, but I so no reason it should stop the rest of the node from responding to an otherwise-valid new head tipset.